Fixed #1218.

daftshady opened the pull request:
Fixed #1218, please review this.
1. Split `If-None-Match` with `,` to handle multiple etag in a single header.
2. If "*" is given and any current entity exists for that resource, this method should return `True`.
3. Find etag match in tags (maybe including weak validator).

daftshady commented:
@bdarnell Can you please review this PR?

bdarnell commented:
* This needs tests, especially the new functionality of `W/` and `*`.
* It's not correct to simply split on commas; a quoted etag value may contain commas (I think; I haven't dug through the specs to see what exactly is allowed).
* RFC 2616 has been replaced; Etags in HTTP/1.1 are now specified in 7232: https://tools.ietf.org/html/rfc7232
* Should the weak validator support be optional? I'm not sure exactly what the use cases are for weak validators but it sounds like sometimes you'd want to ignore them and only use strong matches.
* redbot.org has some tests for etag-related functionality. Consider using or adding to the tests in maint/test/redbot (although I haven't run these tests in a while and it looks like they're a little broken at the moment).

daftshady commented:
@bdarnell Thanks for reviewing. I added another commit.
1. It was not my intention to make weak validator to be optional. I fixed it.
2. I agree that quoted etag value can contain commas although I also haven't found any spec for etag value in RFC 7232. So I used regex instead of simply splitting on commas.
I will add tests to `red_test.py` soon.

daftshady commented:
@bdarnell Please review new commits.
I used weak comparison according to https://tools.ietf.org/html/rfc7232#section-3.2 and added testcases covering newly added wildcard, weak entity to redbot.

bdarnell commented:
Can you merge or rebase your branch so github's automatic merge will work and it can send the branch to travis-ci? This commit fails on Python 3 (using a str regex on a byte value). It would be good to have at least some of the new tests in the main test suite instead of red_test so they are run automatically (it doesn't look like redbot is doing anything interesting with these tests).

daftshady commented:
I fixed the Python 3 issue and rebased my branch.
I also added etag tests to `web_test` so they can run automatically.
